### PR TITLE
[onert] Adding shape inference for Sub and its test case

### PR DESCRIPTION
**Pull request by YiHyunJin**
This adds shape inference for Sub and its test case.
Signed-off-by: YiHyunjin <hj0412.yi@samsung.com>

### CODE REVIEW

File: tests/nnapi/specs/V1_2/sub_dynamic.mod.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+# limitations under the License.
+#
+
+# refer to tanh_v1_dynamic.mod.py about the structore
+
+# This adds reshape as the first op in a model and
+# returns output of reshape, which is dynamic tensor
+
+'''
+Testing Abs op when the input is dynamic.
```
Comment (hasw7569) [inline tests/nnapi/specs/V1_2/sub_dynamic.mod.py:24]: Is it Abs op ?
I think you need to fix abs op -> sub op.

**Comment by hyunsik-yoon** [comment]
Regarding `sub_dynamic.mod.py`,
1. could you rename it to `sub_dynamic_nnfw.mod.py`,
- https://github.com/Samsung/ONE/pull/1380/files#diff-e77c1ab7a3495d9b6dcc45b32525022eR1
- https://github.com/Samsung/ONE/pull/1258#issuecomment-632522153
2. could you change the output shape?
- https://github.com/Samsung/ONE/pull/1380/files#diff-e77c1ab7a3495d9b6dcc45b32525022eR51

**Comment by YiHyunJin** [comment]
@hyunsik-yoon
I renamed to sub_dynamic_nnfw.mod.py.
and modified testcase.

File: tests/nnapi/specs/V1_2/sub_dynamic_nnfw.mod.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+model = Model()
+
+model_input1_shape = [3, 4]   # first input shape of Sub. 12 float32s
+
+dynamic_layer = dynamic_tensor.DynamicInputGenerator(model, model_input1_shape)
+
+test_node_input = dynamic_layer.getTestNodeInput() # first input of Sub is dynamic tensor
+
+i2 = Input("op2", "TENSOR_FLOAT32", "{1, 4}") # second input of Sub. 4 float32s
+t1 = Internal("op3", "TENSOR_FLOAT32", "{1}") # result of first Sub1. dynamic and shape is not known
```
Comment (hyunsik-yoon) [inline tests/nnapi/specs/V1_2/sub_dynamic_nnfw.mod.py:14]: Please use `{}` for a tensor of unknown rank. https://github.com/Samsung/ONE/pull/1318/files

File: tests/nnapi/specs/V1_2/sub_dynamic_nnfw.mod.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# model
```
Comment (hyunsik-yoon) [inline tests/nnapi/specs/V1_2/sub_dynamic_nnfw.mod.py:1]: I guess maybe more meaningful comment would be better.

File: tests/nnapi/specs/V1_2/sub_dynamic_nnfw.mod.py (hunk 2 above)
Comment (hyunsik-yoon) [inline tests/nnapi/specs/V1_2/sub_dynamic_nnfw.mod.py:14]: maybe `Sub1` is typo of `Sub`?

**Comment by hyunsik-yoon** [comment]
Maybe in next PR, you could refactor this one and `Add` since their inference code is same. :-)

**Comment by YiHyunJin** [comment]
@hyunsik-yoon 

> Maybe in next PR, you could refactor this one and Add since their inference code is same. :-)

https://github.com/Samsung/ONE/pull/1419

**Comment by hyunsik-yoon** [comment]
I am not sure what you've changed since last review. 
Maybe simple comment or making multiple commits (not amending one commit multiple times) may better next time. :-)

**Pull request merged**